compiler/noirc_frontend/src/lexer/token.rs

```diff
@@ -9,12 +9,105 @@ use crate::lexer::errors::LexerErrorKind;
 /// smallest unit of grammar. A parser may (will) decide to parse
 /// items differently depending on the Tokens present but will
 /// never parse the same ordering of identical tokens differently.
+#[derive(PartialEq, Eq, Hash, Debug, Clone, PartialOrd, Ord)]
+pub enum Tok<'input> {
```

Comment:
Can we elaborate in a comment on `Tok` vs. `Token`. Also I feel the comment above `Tok` should remain about `Token` ?

Reply:
I think we should rename `Tok` to something clearer as well. Otherwise the names don't hint at how they're different at all. BorrowedToken? Not sure.

Reply:
I came up with:
- `Tok` -> `StrToken`
- `Token` -> `DynToken`

`BorrowedToken` seems to be the clearest so leaning towards renaming `Tok` to that.